modelon/impact/client/entities/case.py

```diff
@@ -430,23 +492,49 @@ def get_trajectories(self) -> Result:
             self._sal,
         )
 
-    def get_artifact(self, artifact_id: str) -> Tuple[bytes, str]:
+    def get_artifact(self, artifact_id: str) -> CustomArtifact:
         """
```

Comment:
If the user knows artifactId then the extra call to get list of artefacts is unnecessary overhead. For incorrect artifact there'll exception when trying to download. To satisfy both:
```
def get_artifact(self, artifact_id: str, check_id = True)
```
If `check_id` is True the function checks if the artifact is present on the server. If the argument is False no remote calls are done and exception may occur when downloading the artifact.

Reply:
Not really. `download_as` can be None up until we actually need to download. Other alternative is to have:
```
def get_artifact(self, artifact_id: str, download_as = None)
```
and skip calling remote if `download_as` is specified. That would also cover the case of multisim when the client sets different file names when downloading artifacts for each case instead of creating separate directories per case.

Reply:
Thanks for the suggestions @iakovn ! I have now pushed the suggested fixes :)